rudderstack/integrations/destinations/destination_snowflake_test.go

```diff
@@ -0,0 +1,228 @@
+package destinations_test
+
+import (
+	"testing"
+
+	cmt "github.com/rudderlabs/terraform-provider-rudderstack/internal/testutil/cm"
+	c "github.com/rudderlabs/terraform-provider-rudderstack/rudderstack/configs"
+)
+
+func TestDestinationResourceSnowflake(t *testing.T) {
+	cmt.AssertDestination(t, "snowflake", []c.TestConfig{
+		{
+			TerraformCreate: `
+				account = "example-account"
+				database = "example-database"
+				warehouse = "example-warehouse"
+				user = "example-user"
+				password = "example-password"
+				use_rudder_storage = true
+				sync {
+					frequency = "30"
+				}
+			`,
+			APICreate: `{
+				"account": "example-account",
+				"database": "example-database",
+				"warehouse": "example-warehouse",
+				"user": "example-user",
+				"password": "example-password",
+				"syncFrequency": "30",
+				"useRudderStorage": true,
+				"additionalProperties": true
+			}`,
+			TerraformUpdate: `
+			account = "example-account"
+			database = "example-database"
+			warehouse = "example-warehouse"
+			user = "example-user"
+			password = "example-password"
+			use_rudder_storage = false
+			sync {
+				frequency = "60"
+				start_at                  = "10:00"
+				exclude_window_start_time = "11:00"
+				exclude_window_end_time   = "12:00"
+			}
+			namespace = "example-namespace"
+			json_paths = "./example-paths"
+			prefix = "example-prefix"
+      s3 {
+				 bucket_name = "example-bucket-name"
+				 access_key_id = "example-access-key-id"
+				 access_key = "example-access-key"
+				 enable_sse = true
+				}
+			`,
+			APIUpdate: `{
+				"account": "example-account",
+				"database": "example-database",
+				"warehouse": "example-warehouse",
+				"user": "example-user",
+				"password": "example-password",
+				"namespace": "example-namespace",
+				"syncFrequency": "60",
+				"syncStartAt": "10:00",
+				"excludeWindow": {
+					"excludeWindowStartTime": "11:00",
+					"excludeWindowEndTime": "12:00"
+				},
+				"useRudderStorage": false,
+				"additionalProperties": true,
+				"jsonPaths": "./example-paths",
+				"cloudProvider": "AWS",
+				"prefix": "example-prefix",
+        "bucketName": "example-bucket-name",
+        "accessKeyID": "example-access-key-id",
+        "accessKey": "example-access-key",
+        "enableSSE": true
```

Comment:
Can we fix the formatting

Reply:
let's add more test cases? for other storage providers?

Reply:
yeah, I have added for all the three providers AWS, GCP and azure